NPE in VPackSerializers when serializing ArangoSearch properties

## Problem
`NullPointerException` is thrown when serializing `StoredValue` with no compression specified:
```
Exception in thread "main" com.arangodb.ArangoDBException: com.arangodb.velocypack.exception.VPackParserException: java.lang.NullPointerException
	at com.arangodb.internal.util.ArangoSerializerImpl.serialize(ArangoSerializerImpl.java:82)
	at com.arangodb.internal.util.ArangoSerializerImpl.serialize(ArangoSerializerImpl.java:51)
	at com.arangodb.internal.util.DefaultArangoSerialization.serialize(DefaultArangoSerialization.java:47)
	at com.arangodb.internal.InternalArangoDatabase.createArangoSearchRequest(InternalArangoDatabase.java:470)
	at com.arangodb.async.internal.ArangoDatabaseAsyncImpl.createArangoSearch(ArangoDatabaseAsyncImpl.java:460)
	at za.co.absa.spline.persistence.ArangoManagerImpl.$anonfun$createSearchViews$1(ArangoManager.scala:255)
	at scala.concurrent.Future$.$anonfun$traverse$1(Future.scala:850)
	at scala.collection.TraversableOnce$folder$1.apply(TraversableOnce.scala:196)
	at scala.collection.TraversableOnce$folder$1.apply(TraversableOnce.scala:194)
	at scala.collection.immutable.Set$Set2.foreach(Set.scala:181)
	at scala.collection.TraversableOnce.foldLeft(TraversableOnce.scala:199)
	at scala.collection.TraversableOnce.foldLeft$(TraversableOnce.scala:192)
	at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:108)
	at scala.concurrent.Future$.traverse(Future.scala:850)
	at za.co.absa.spline.persistence.ArangoManagerImpl.createSearchViews(ArangoManager.scala:253)
	at za.co.absa.spline.persistence.ArangoManagerImpl.$anonfun$initialize$9(ArangoManager.scala:77)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: com.arangodb.velocypack.exception.VPackParserException: java.lang.NullPointerException
	at com.arangodb.velocypack.VPack.serialize(VPack.java:860)
	at com.arangodb.velocypack.VPack.serialize(VPack.java:847)
	at com.arangodb.internal.util.ArangoSerializerImpl.serialize(ArangoSerializerImpl.java:77)
	... 24 more
Caused by: java.lang.NullPointerException
	at com.arangodb.internal.velocypack.VPackSerializers.lambda$static$13(VPackSerializers.java:237)
	at com.arangodb.velocypack.VPack.addValue(VPack.java:930)
	at com.arangodb.velocypack.VPack.serializeField(VPack.java:912)
	at com.arangodb.velocypack.VPack.serializeFields(VPack.java:890)
	at com.arangodb.velocypack.VPack.serializeObject(VPack.java:878)
	at com.arangodb.velocypack.VPack.addValue(VPack.java:964)
	at com.arangodb.velocypack.VPack.serialize(VPack.java:858)
	... 26 more
```

## Test case

```java
VPackSerializers.ARANGO_SEARCH_PROPERTIES.serialize(
    new VPackBuilder(),
    "foo",
    new ArangoSearchProperties() {{
        addStoredValues(new StoredValue(List.of("bar")));
    }},
    (builder, attribute, entity) -> {
    }
);
```

--- a/src/main/java/com/arangodb/internal/velocypack/VPackSerializers.java
+++ b/src/main/java/com/arangodb/internal/velocypack/VPackSerializers.java
@@ -234,7 +234,9 @@ public class VPackSerializers {
                     builder.add(field);
                 }
                 builder.close();
-                builder.add("compression", storedValue.getCompression().getValue());
+                if (storedValue.getCompression() != null) {
+                    builder.add("compression", storedValue.getCompression().getValue());
+                }
                 builder.close(); // close object
             }
             builder.close(); // close array

--- a/src/test/java/com/arangodb/internal/velocypack/VPackSerializersTest.java
+++ b/src/test/java/com/arangodb/internal/velocypack/VPackSerializersTest.java
@@ -1,0 +1,81 @@
+/*
+ * DISCLAIMER
+ *
+ * Copyright 2016 ArangoDB GmbH, Cologne, Germany
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright holder is ArangoDB GmbH, Cologne, Germany
+ */
+
+package com.arangodb.internal.velocypack;
+
+import com.arangodb.entity.ViewType;
+import com.arangodb.entity.arangosearch.ArangoSearchCompression;
+import com.arangodb.entity.arangosearch.ArangoSearchProperties;
+import com.arangodb.entity.arangosearch.StoredValue;
+import com.arangodb.model.arangosearch.ArangoSearchCreateOptions;
+import com.arangodb.velocypack.VPack;
+import com.arangodb.velocypack.VPackSlice;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class VPackSerializersTest {
+
+    private VPack vpack;
+
+    @Before
+    public void init() {
+        vpack = new VPack.Builder()
+                .registerSerializer(ArangoSearchProperties.class, VPackSerializers.ARANGO_SEARCH_PROPERTIES)
+                .build();
+    }
+
+    @Test
+    public void serializeArangoSearchProperties() {
+        final ArangoSearchCreateOptions opts = new ArangoSearchCreateOptions()
+                .storedValues(new StoredValue(Collections.singletonList("dummy"), ArangoSearchCompression.lz4));
+
+        final VPackSlice slice = vpack.serialize(opts);
+
+        assertThat(slice.isObject(), is(true));
+        assertThat(slice.get("type").isString(), is(true));
+        assertThat(slice.get("type").getAsString(), is(ViewType.ARANGO_SEARCH.name()));
+        assertThat(slice.get("storedValues"), notNullValue());
+        assertThat(slice.get("storedValues").isArray(), is(true));
+        assertThat(slice.get("storedValues").size(), is(1));
+        assertThat(slice.get("storedValues").get(0).isObject(), is(true));
+        assertThat(slice.get("storedValues").get(0).get("fields").isArray(), is(true));
+        assertThat(slice.get("storedValues").get(0).get("fields").size(), is(1));
+        assertThat(slice.get("storedValues").get(0).get("fields").get(0).isString(), is(true));
+        assertThat(slice.get("storedValues").get(0).get("fields").get(0).getAsString(), is("dummy"));
+        assertThat(slice.get("storedValues").get(0).get("compression").isString(), is(true));
+        assertThat(slice.get("storedValues").get(0).get("compression").getAsString(), is(ArangoSearchCompression.lz4.name()));
+    }
+
+    @Test
+    public void serializeArangoSearchPropertiesWithDefaultCompression() {
+        final ArangoSearchCreateOptions opts = new ArangoSearchCreateOptions()
+                .storedValues(new StoredValue(Collections.singletonList("dummy")));
+
+        final VPackSlice slice = vpack.serialize(opts);
+
+        assertThat(slice.get("storedValues").get(0).get("compression").isNone(), is(true));
+    }
+}

